cmd/common/oci.go

```diff
@@ -256,12 +261,57 @@ func NewRunCommand(rootCmd *cobra.Command, runtime runtime.Runtime, hiddenColumn
 
 		timeoutDuration := time.Duration(timeoutSeconds) * time.Second
 
-		imageName := args[0]
+		var imageName string
+		if len(args) > 0 {
+			imageName = args[0]
+		}
+
```

Comment:
What about if the user runs `... run -f foo.yaml myimage:latest`? I think it should fail.

Reply:
I don't know - should it?

I could imagine users having setup filters for namespaces / pods etc in a manifest and then want to run different gadgets using it.

Reply:
IMO it's too confusing to have this mixed operation mode. What about if there are multiple instances on the manifest? What about if the manifest has an image defined but it doesn't match with the one on the CLI?

Reply:
IMO, it breaks the balance between flexibility and complexity, so it should fail ... "Keep it simple" 🙂

Reply:
I don't know. I don't like artificially preventing users from doing things that would work just fine by intuition (well, at least my intuition 😄)

Currently, everything specified on the command-line would take precedence over what is written in the manifest. Even for multiple instances that would be true.

But I seem to be in the minority seeing your comments + upvotes, hehe.

So, just so that we're on the same page, please confirm:
* using `-f` should prevent specifying an image on CLI
* parameters specified on the CLI should be ignored (or warned about? tricky.) when using `-f`

Careful, though: this would (without introducing exceptions) also apply to connection related settings (like setting the `--remote-address`, for example).

Reply:
Perhaps I'm misunderstanding something. Is it possible to specify things like `--remote-address`, `--connection-timeout` (and other global parameters) on the manifest file? I would expect this not to be possible as global params **aren't** related to a specific gadget instance. 

For "instance parameters", I think they won't be available as not gadget image is specified.  Hence, in my understanding there is not any parameter that can be overwritten:
- global params: only available on the CLI 
- instance params: only available on the manifest

Reply:
You're totally right about the global params, I mixed those up.

Instance params _are_ available, but only if a gadget image is specified - but I will change it to the behavior you described.